scripts/process_results.py

```
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
+"""
+Aggregate BayesPrism results (simulation / real / per-donor).
+
+Usage examples
+--------------
+# simulation (ground truth available)
+python process_results.py --dataset=PBMC --simulation
+
+# real bulk deconvolution
+python process_results.py --dataset=Real_ADP
+
+# per-donor references
+python process_results.py --dataset=Real_ADP --perdonor
+"""
+from __future__ import annotations
+import argparse, glob, os, re, sys
+import numpy as np, pandas as pd
+from scipy.stats import pearsonr
+from sklearn.metrics import root_mean_squared_error
+
+# --------------------------------------------------------------------
+#  Arguments
+# --------------------------------------------------------------------
+parser = argparse.ArgumentParser(description="Aggregate BayesPrism outputs")
+parser.add_argument(
+    "--dataset",
+    required=True,
+    help="Dataset name (folder under ../data/deconvolution/)",
+)
+parser.add_argument(
+    "--simulation",
+    action="store_true",
+    help="Synthetic pseudobulks with ground-truth proportions.",
+)
+parser.add_argument(
+    "--perdonor",
+    action="store_true",
+    help="One-reference-per-donor experiment (no ground truth).",
+)
+args = parser.parse_args()
+
+# --------------------------------------------------------------------
+# Settings (add your transform name here!)
+# --------------------------------------------------------------------
+valid_transforms = {
+    "rawSN",
+    "pcaSN",
+    "degSN",
+    "degRandSN",
+    "degIntSN",
+    "degOtherSN",
+    "degPCA_SN",
+    "degpcaSN",
+    "scviSN",
+    "scvi_LSshift_SN",
+    "degScviSN",
+    "degScviLSshift_SN",
+}
+
+# -------- mutually-exclusive flags --------
+if args.simulation and args.perdonor:
+    sys.exit("ERROR: Use only one of --simulation or --perdonor.")
```

I love seeing the pythonic "fail fast" philosophy at work :)

---

👯